backend/infrahub/core/schema/definitions/core/resource_pool.py

```diff
@@ -32,6 +32,17 @@
     ],
 )
 
+core_weighted_pool_resource = GenericSchema(
+    name="WeightedPoolResource",
+    namespace="Core",
+    label="Weighted Pool Resource",
+    description="Resource to be used in a pool, its weight is used to determine its priority on allocation.",
+    include_in_menu=False,
+    branch=BranchSupportType.AWARE,
+    generate_profile=False,
+    attributes=[Attr(name="weight", kind="Number", optional=True, order_weight=10000)],
```

Comment:
Maybe we want to have a different name here such as `allocation_weight` or something similar to avoid conflicting with a potential user defined attribute.